src/main/java/core/basesyntax/Car.java

```diff
@@ -1,13 +1,77 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
-/**
- * Make this class immutable. See requirements in the README file
- */
-public class Car {
-    private int year;
-    private String color;
-    private List<Wheel> wheels;
-    private Engine engine;
+public final class Car {
+    private final int year;
+    private final String color;
+    private final List<Wheel> wheels;
+    private final Engine engine;
+
+    public Car(int year, String color, List<Wheel> wheels, Engine engine) {
+        this.year = year;
+        this.color = color;
+        this.engine = (engine == null ? null : engine.clone());
+        this.wheels = cloneWheels(wheels);
+    }
+
+    public Car changeEngine(Engine newEngine) {
+        return new Car(year, color, getWheels(), newEngine);
```

Comment:
```suggestion
        return new Car(year, color, wheels, newEngine);
```
you clone wheels in the constructor and in method getWheels